### PR TITLE
Show information about retracted versions in "pub outdated"

### PR DESCRIPTION
**Pull request by szakarias**
Fixes https://github.com/dart-lang/pub/issues/4033

### CODE REVIEW

File: test/testdata/goldens/outdated/outdated_test/circular dependency on root.txt [hunk 1]
```diff
@@ -124,6 +125,7 @@ $ pub outdated --json --no-dev-dependencies
       "package": "foo",
       "kind": "direct",
       "isDiscontinued": false,
+      "isRetracted": false,
```
Comment (sigurdm) [inline test/testdata/goldens/outdated/outdated_test/circular dependency on root.txt:128]: This looks a bit wrong - should it not be the versions knowing if they are retreacted?

**Review by sigurdm — COMMENTED** [review]
I think we should modify the filtering logic, such that rows where the current version is retracted (but equal to latests) is shown

**Review by sigurdm — APPROVED** [review]

File: test/testdata/goldens/outdated/outdated_test/circular dependency on root.txt (hunk 1 above)
Comment (sigurdm) [reply test/testdata/goldens/outdated/outdated_test/circular dependency on root.txt]: Renamed this is much better

Comment (szakarias) [reply test/testdata/goldens/outdated/outdated_test/circular dependency on root.txt]: Renamed this to isCurrentRetracted to clarify.

**Comment by szakarias** [comment]
> I think we should modify the filtering logic, such that rows where the current version is retracted (but equal to latests) is shown

We already show these, aka when the newest version is retracted. The concept of `Latest` does not apply to retracted versions, that is the latest version cannot be a retracted version.

**Pull request merged**